service/worker/scheduler/workflow.go

```diff
@@ -1234,15 +1235,18 @@ func (s *scheduler) processBuffer() bool {
 			tryAgain = true
 			continue
 		}
-		result, err := s.startWorkflow(start, req)
 		metricsWithTag := s.metrics.WithTags(map[string]string{
 			metrics.ScheduleActionTypeTag: metrics.ScheduleActionStartWorkflow})
+		metricsWithTag.Counter(metrics.ScheduleActionAttempt.Name()).Inc(1)
+		result, err := s.startWorkflow(start, req)
+
 		if err != nil {
 			s.logger.Error("Failed to start workflow", "error", err)
-			metricsWithTag.Counter(metrics.ScheduleActionErrors.Name()).Inc(1)
-			// TODO: we could put this back in the buffer and retry (after a delay) up until
-			// the catchup window. of course, it's unlikely that this workflow would be making
-			// progress while we're unable to start a new one, so maybe it's not that valuable.
+
+			var workflowExecutionAlreadyStarted *serviceerror.WorkflowExecutionAlreadyStarted
+			if !errors.As(err, &workflowExecutionAlreadyStarted) {
```

Comment:
did you check that this actually works? the error gets serialized+wrapped by go sdk when it crosses the activity/workflow boundary and the I don't think the type will be reconstituted like this

Reply:
to my best knowledge, and looking at the other examples in the code - yes, it should work. This is serviceerror.

Reply:
can you test it to make sure? ideally integration test (would be easy with a counter in ScheduleInfo). but manual test is okay

Reply:
tested. it is ApplicationError, and the actual errors needs to be Unwrapped.

Reply:
and tested again. we have applicatinoError inside ApplicationError, with WorkflowExecutionAlreadyStarted being Type of that internal error, and "" being a type of outer error